scripts/openduck.py

```diff
@@ -0,0 +1,745 @@
+    report.add_argument('-i', '--iterations', default=20, type=int, help='Number of bootstrapping iterations for jarzynski analysis.')
+    report.add_argument('-t', '--step-threshold', default=2500, type=int, help='Steps_treshold to find the minima.')
+    report.add_argument('-f', '--format', type=str.lower, default='amber', choices=('amber', 'openmm'), help='Data where the results come from. Amber by default.')
+
+    #Arguments for chunk
+    chunk = modes.add_parser('chunk', help='Chunk a protein for Dynamic Undocking.')
+    chunk.set_defaults(mode='Chunk')
+    chunk.add_argument('-y', '--yaml-input', type=str, default=None, help='nput yaml file with the all the arguments for chunking.')
+    chunk.add_argument('-l', '--ligand', type=str, default = None, help='ligand mol file to use as reference for interaction.')
+    chunk.add_argument('-i', '--interaction', type=str, default = None, help='Protein atom to use for ligand interaction.')
+    chunk.add_argument('-r', '--receptor', type=str, default = None, help='Protein pdb file to chunk, or chunked protein if mode is "for_chunk".')
+    chunk.add_argument('-c', '--cutoff', type=float, default = 9, help='Cutoff distance to define chunk.')
+    chunk.add_argument('-b', '--ignore-buffers', action='store_true', help='Do not remove buffers (solvent, ions etc.)')
+    chunk.add_argument('-o', '--output', type=str, default='protein_out.pdb',help='Output format for the chunked protein receptor.')
+
+    args = args_sanitation(parser, modes)
+
+    return args, parser
+
+def duck_smd_runs(input_checkpoint, pickle, num_runs, md_len, gpu_id, start_dist, init_velocity, save_dir, wqb_threshold=None):
+    """
+    Run molecular dynamics and steered molecular dynamics simulations on a system specified by an input checkpoint and a pickle file.
+    
+    Args: 
+        input_checkpoint (str): The path to the input checkpoint file.
+        pickle (str): The path to the pickle file.
+        num_runs (int): The number of runs to perform.
+        md_len (float): The length of each MD run in picoseconds.
+        gpu_id (int): The ID of the GPU to use.
+        start_dist (float): The starting distance for the steered MD simulations.
+        init_velocity (float): The initial velocity for the steered MD simulations.
+        save_dir (str): The directory to save the output files.
+        wqb_threshold (float): An optional threshold for the work required to break a bond. If specified and the work obtained is lower than the threshold, the function will terminate and return the wqb. Default is None.
+    
+    Returns:
+       If the `wqb_threshold` is reached, the function returns the `wqb` value. Otherwise, the function returns None.
+    """
+    from duck.steps.normal_md import perform_md
+    from duck.steps.steered_md import run_steered_md
+    from duck.utils.check_system import check_if_equlibrated
+    from duck.utils.analysis_and_report import get_Wqb_value
+    import filecmp
```

Comment:
It's more standard to put imports at the start of the file (I know this was not really followed in the past)

Reply:
I am not sure on this one. I agree that the imports shared across the script should be at the start of the file.
However, as the openduck.py script serves as a way to order the different executions, I felt it was unnecessary to load everything when only a couple libraries will be used.
Instead, the libraries are loaded only when entering the subcommand that will use them. 

I can change it to have everything at the top, but i am not convinced is the best way

Reply:
No problem, let's leave it like this.